### PR TITLE
basic golangci-lint

### PR DESCRIPTION
**Pull request by faddat**
this isn't done yet, and we'll have some questions.  

I expect that this will throw a bunch of errors, which don't exactly exist, so we might want to maybe exclude some files, like attestation.go

```json
{
    "makefile.extensionOutputFolder": "./.vscode",
    "go.inferGopath": false,
    "go.buildTags": "secretcli"
}
```

^ this was very helpful for editor config, and will also make golangci-lint work approximately correctly

### CODE REVIEW

File: cmd/secretd/attestation.go [hunk 1]
```diff
@@ -492,6 +492,9 @@ Please report any issues with this command
 			}`, base64.StdEncoding.EncodeToString(cert)))
 
 			resp, err := http.Post(fmt.Sprintf(`%s`, regUrl), "application/json", bytes.NewBuffer(data))
+			if err != nil {
+				log.Fatalln(err)
+			}
```
Comment (assafmo) [inline cmd/secretd/attestation.go:497]: Nice catch

File: x/registration/remote_attestation/remote_attestation.go [hunk 2]
```diff
@@ -184,7 +184,6 @@ func verifyAttReport(attnReportRaw []byte, pubK []byte) ([]byte, error) {
 				if err != nil && len(platInfo) != 105 {
 					return nil, errors.New("illegal PlatformInfoBlob")
 				}
-				platInfo = platInfo[4:]
```
Comment (assafmo) [inline x/registration/remote_attestation/remote_attestation.go:187]: Why?

**Comment by assafmo** [comment]
Wow, great work @faddat! A lot of bad error handling on our part. :eyes:
This will probably break some tests...

**Comment by faddat** [comment]
actually I was wondering about that part (breaking tests)

I don't know if I have my editor configured quite right yet.

File: x/registration/remote_attestation/remote_attestation.go (hunk 2 above)
Comment (faddat) [reply x/registration/remote_attestation/remote_attestation.go:187]: never used.

I questioned it too, an d checked, and all the things seemed ok?

wdyt?

**Comment by faddat** [comment]
Quick question, can I use the SGX SDK from my precious M1 Mac which I could never give up?

**Comment by Cashmaney** [comment]
> Quick question, can I use the SGX SDK from my precious M1 Mac which I could never give up?
> 
> ya know also I can make a devenv docker image for ya guys 1 sec

We haven't managed to get an image for M1/arm processors working. Even with cross compiling (e.g. https://github.com/scrtlabs/SecretNetwork/blob/cross-compile-localsecret--to-arm64/deployment/docker/localsecret-docker-bake.hcl) we haven't found any combination of container/emulator that will get things working properly. I'm not entirely sure if it's due to some compilation flag or something else - we need to take some time and try to investigate/zoom in on the exact issue (and if it's at all possible). Feel free to take a crack at it - any progress or insight will be very much appreciated

**Comment by faddat** [comment]
hmmmmmm

will poke.

must m1 mac....

**Comment by assafmo** [comment]
This is actually a very big deal for us if you can find a lead on LocalSecret for M1

**Comment by faddat** [comment]
So, I'll roll through and maybe exclude some things from the linter.

**Comment by faddat** [comment]
I think that you've found the way to do localsecret on m1: gitpod.

This PR now updates gitpod some, too.

**Comment by faddat** [comment]
I think it's set :)

File: x/registration/remote_attestation/remote_attestation.go (hunk 2 above)
Comment (assafmo) [reply x/registration/remote_attestation/remote_attestation.go:187]: Yeah an oversight probably :confused:

File: app/app.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ import (
 	porttypes "github.com/cosmos/ibc-go/v3/modules/core/05-port/types"
 	ibchost "github.com/cosmos/ibc-go/v3/modules/core/24-host"
 	ibckeeper "github.com/cosmos/ibc-go/v3/modules/core/keeper"
-	v1_3 "github.com/enigmampc/SecretNetwork/app/upgrades/v1.3"
+	v1_3 "github.com/enigmampc/SecretNetwork/app/upgrades/1point3"
```
Comment (assafmo) [inline app/app.go:40]: Is this a linter issue?

File: app/app.go [hunk 4]
```diff
@@ -210,6 +210,8 @@ func (app *SecretNetworkApp) GetScopedIBCKeeper() capabilitykeeper.ScopedKeeper
 	return app.ScopedIBCKeeper
 }
 
+// TODO: this doesn't look right - Jacob
+//nolint:staticcheck // we are doing this to satisfy an ibc requirement.  Likely possible to fix it...
 func (app *SecretNetworkApp) GetTxConfig() client.TxConfig {
 	return app.GetTxConfig()
 }
```
Comment (assafmo) [inline app/app.go:217]: What should we do here?

File: client/docs/static/openapi/openapi.yaml [hunk 5]
```diff
@@ -28473,7 +28473,7 @@ paths:
         required: true
       responses:
         "202":
-          description: Tx was succesfully generated
+          description: Tx was successfully generated
```
Comment (assafmo) [inline client/docs/static/openapi/openapi.yaml:28476]: I think that this file is auto-generated. We should find the source and fix it there

File: cmd/secretd/root.go [hunk 6]
```diff
@@ -330,76 +306,3 @@ func exportAppStateAndTMValidators(
...
-
-		return nil
-	}
-
-	cmd.RunE = wrappedFunc
-	return cmd
-}
-
-func initConfig(ctx *client.Context, cmd *cobra.Command) error {
-	cmd.PersistentFlags().Bool(flagLegacyHdPath, false, "Flag to specify the command uses old HD path - use this for ledger compatibility")
```
Comment (assafmo) [inline cmd/secretd/root.go:363]: Where all of this is going to?

File: go-cosmwasm/api/callbacks_cgo_mock.go [hunk 7]
```diff
@@ -1,61 +0,0 @@
-//go:build secretcli
```
Comment (assafmo) [inline go-cosmwasm/api/callbacks_cgo_mock.go:1]: Not sure we can delete this file

File: x/registration/client/cli/query.go [hunk 8]
```diff
@@ -46,17 +43,17 @@ func GetCmdEncryptedSeed() *cobra.Command {
...
 			}
 
-			route := fmt.Sprintf("custom/%s/%s/%s", types.QuerierRoute, keeper.QueryEncryptedSeed, nodeId)
+			route := fmt.Sprintf("custom/%s/%s/%s", types.QuerierRoute, keeper.QueryEncryptedSeed, nodeID)
 			res, _, err := clientCtx.Query(route)
 			if err != nil {
 				return err
 			}
-			fmt.Println(fmt.Sprintf("0x%s", hex.EncodeToString(res)))
+			fmt.Printf(fmt.Sprintf("0x%s", hex.EncodeToString(res))) //nolint:staticcheck
```
Comment (assafmo) [inline x/registration/client/cli/query.go:56]: Missing linefeed?

File: x/registration/client/cli/query.go [hunk 9]
```diff
@@ -106,46 +103,3 @@ func GetCmdMasterParams() *cobra.Command {
 	flags.AddQueryFlagsToCmd(cmd)
 	return cmd
 }
-
```
Comment (assafmo) [inline x/registration/client/cli/query.go:109]: Why is this being deleted?

File: x/registration/remoteAttestation/types.go [hunk 10]
```diff
@@ -49,39 +44,3 @@ func parseReport(quoteBytes []byte, quoteHex string) *QuoteReportData {
 	qrData.ReportBody.ReportData = quoteHex[736:864]
 	return qrData
 }
-
```
Comment (assafmo) [inline x/registration/remoteAttestation/types.go:52]: Is this the linter saying that it's unused?

File: x/registration/remoteAttestation/remote_attestation.go [hunk 11]
```diff
@@ -184,14 +181,7 @@ func verifyAttReport(attnReportRaw []byte, pubK []byte) ([]byte, error) {
 				if err != nil && len(platInfo) != 105 {
 					return nil, errors.New("illegal PlatformInfoBlob")
 				}
-				platInfo = platInfo[4:]
-
-				//piBlob := parsePlatform(platInfo)
```
Comment (assafmo) [inline x/registration/remoteAttestation/remote_attestation.go:189]: I think we want this for debugging porpuses?

https://github.com/scrtlabs/SecretNetwork/pull/954/files#r928222109

**Review by assafmo — COMMENTED** [review]
A lot of code sections being deleted here. Is this the linter saying that they're unused?

**Comment by assafmo** [comment]
For example this https://github.com/scrtlabs/SecretNetwork/pull/954#discussion_r928220868 adds CLI Ledger support, as Secret is using coin type 529. It's probably not tested in the CI, but it's definitely used somewhere... I think

File: app/app.go (hunk 3 above)
Comment (faddat) [reply app/app.go:40]: yep

File: app/app.go (hunk 4 above)
Comment (faddat) [reply app/app.go:217]: good question!

1 moment....

File: cmd/secretd/root.go (hunk 6 above)
Comment (faddat) [reply cmd/secretd/root.go:363]: better question: what is it for?

It was showing as unused, that's why it was taken out.

File: x/registration/remoteAttestation/remote_attestation.go (hunk 11 above)
Comment (faddat) [reply x/registration/remoteAttestation/remote_attestation.go:189]: ahhh ok :)

**Comment by faddat** [comment]
Hey, I am just running through and updating some things.  

This one may take me another few days to push over the line, as I am still in Seoul.  

I think when I first wrote it I didn't understand that a lot of commented out code in here, actually runs.

**Comment by faddat** [comment]
* #1064 Is much better, so I will close this.

**Pull request closed without merging**